contracts/dev-utils/contracts/src/OrderValidationUtils.sol

```diff
@@ -126,21 +126,20 @@ contract OrderValidationUtils is
             transferableTakerAssetAmount
         );
 
-        // Execute the maker transfers.
-        fillableTakerAssetAmount = getSimulatedOrderMakerTransferResults(
-            order,
-            order.takerAddress,
-            fillableTakerAssetAmount
-        ) == OrderTransferResults.TransfersSuccessful ? fillableTakerAssetAmount : 0;
-
-        if (!_isAssetDataValid(order.takerAssetData)) {
-            fillableTakerAssetAmount = 0;
-        }
 
-        if (order.takerFee != 0 && !_isAssetDataValid(order.takerFeeAssetData)) {
+        // Ensure that all of the asset data is valid. Fee asset data only needs
+        // to be valid if the fees are nonzero.
+        if (
```

Comment:
Can this be rolled into a `areOrderAssetDatasValid` function? Seems a bit more maintainable.